lib/src/Pos/Launcher/Resource.hs

```diff
@@ -163,16 +170,22 @@ allocateNodeResources np@NodeParams {..} sscnp txpSettings initDB = do
             , nrSscState = sscState
             , nrTxpState = txpVar
             , nrDlgState = dlgVar
+            , nrJsonLogConfig = jsonLogConfig
             , ..
             }
 
 -- | Release all resources used by node. They must be released eventually.
 releaseNodeResources ::
        NodeResources ext -> Production ()
-releaseNodeResources NodeResources {..} = do
-    whenJust nrJLogHandle (liftIO . hClose)
-    closeNodeDBs nrDBs
-    releaseNodeContext nrContext
+releaseNodeResources NodeResources {..} =
+    case nrJsonLogConfig of
+        JsonLogDisabled -> return ()
+        JsonLogConfig mVarHandle _ -> do
+            h <- takeMVar mVarHandle
+            (liftIO . hClose) h
+            putMVar mVarHandle h
+            closeNodeDBs nrDBs
```

Comment:
This should be outside of the `case`, right? Otherwise we will only close the DB and release the context if JSON logging is active.

Reply:
Yes. Fixed.